From: Gustavo Noronha Silva <kov@debian.org>
Date: Tue, 17 Jul 2018 16:56:51 +0300
Subject: _check_newline

## Description: Checks if buffer contains only newline before discarding it
## Origin/Author: Torsten Spindler (Canonical)
## Bug: http://pad.lv/298217
---
 libgksu/libgksu.c | 14 ++++++++++----
 1 file changed, 10 insertions(+), 4 deletions(-)

diff --git a/libgksu/libgksu.c b/libgksu/libgksu.c
index 56edae8..9c8305a 100644
--- a/libgksu/libgksu.c
+++ b/libgksu/libgksu.c
@@ -2857,16 +2857,22 @@ gksu_sudo_fuller (GksuContext *context,
 	  /* ignore the first newline that comes right after sudo receives
 	     the password */
 	  fgets (buffer, 255, fdfile);
-	  /* this is the status we are interested in */
-	  fgets (buffer, 255, fdfile);
+          if (!strcmp (buffer, "\n")) 
+            {
+	      /* this is the status we are interested in */
+	      fgets (buffer, 255, fdfile);
+            }
 #else
           fcntl(parent_pipe[0], F_SETFL, fcntl(parent_pipe[0], F_GETFL) & ~O_NONBLOCK);
 
 	  /* ignore the first newline that comes right after sudo receives
 	     the password */
 	  fgets (buffer, 255, infile);
-	  /* this is the status we are interested in */
-	  fgets (buffer, 255, infile);
+          if (!strcmp (buffer, "\n")) 
+            {
+	      /* this is the status we are interested in */
+	      fgets (buffer, 255, infile);
+            }
 #endif
 	}
       else
